overlord/configstate/configcore/proxy.go

```diff
@@ -94,3 +94,19 @@ func validateProxyStore(tr Conf) error {
 	}
 	return err
 }
+
+func validateProxyLimits(tr Conf) error {
+	// pam_env (that reads /etc/environment) has a size limit for
+	// the environment vars of 1024 byte. We need to honor this.
+	for _, key := range []string{"http", "https", "ftp"} {
+		s, err := coreCfg(tr, "proxy."+key)
+		if err != nil {
+			return err
+		}
+		if len(s) > 1024 {
```

Comment:
Is is the number of bytes that we need to track? I suspect you want the UTF-8 encoded byte array that you will then measure instead of the string. Some malicious input could get us to be under 1024 characters and way over 1024 bytes.

Reply:
`len([]byte(s))` should do the trick

Reply:
`len([]byte(s))` is always `len(s)`. Maybe you meant `len([]rune(s))`? But that's not right (or rather, it's correct if `s` is known to be valid utf8).

If the input should be valid UTF-8 we should've checked as it came in, with e.g. `utf8.ValidString` (otherwise the string is just a fancy immutable `[]byte`). Assuming we did that, and then wanted to limit the output to 1024 runes, `utf8.RuneCountInString` would do the thing without copying the data around.

Reply:
Looks like len() returns the byte count already. I wrongly assumed that you need to `[]bytes(.)`, but it's not necessary. If we're only interested in byte count, the code should be ok as it is.

Just for the reference, `len("⌘")` is 3 but `len([]rune("⌘"))` is 1.

Reply:
Ah, that's ... unexpected to say the least.

Still +1 on the patch.

Reply:
@zyga unexpected is what happens when you `range` over a `string` :-)